### PR TITLE
Implemented Patient Education feature

### PR DESCRIPTION
**Pull request by lsivasothy**
screen shot 2018-03-06 at 1 26 09 pm
screen shot 2018-03-06 at 1 26 20 pm
screen shot 2018-03-06 at 1 26 26 pm
screen shot 2018-03-06 at 1 26 36 pm

### CODE REVIEW

**Comment by jpatricia** [comment]
@lakshisiva i feel like the search button is supposed to be in the middle of the page instead of on the right side

File: ChemoTracker/components/PatientEducation/PatientEducation.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          />
+          <Button 
+            style={searchStyles.submitButton} 
+            textStyle={{color: '#fff'}}
+            isDisabled={ !this.state.question }
+            onPressIn={ this.handleQuestionSubmit }>
+              Search
+          </Button>
+        </View>
+        {/* <View style={ searchStyles.section }>
```
Comment (jpatricia) [inline ChemoTracker/components/PatientEducation/PatientEducation.js]: should just delete this if not going to use it anymore

File: ChemoTracker/components/PatientEducation/Search.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+import { ActivityIndicator, FlatList, Linking, ListView, Text, TextInput, TouchableOpacity, View } from 'react-native';
+import { List, ListItem } from "react-native-elements";
+import { TabNavigator } from 'react-navigation';
+import { WebBrowser } from 'expo';
+import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
+import IonIcon from 'react-native-vector-icons/Ionicons';
+import { searchStyles } from '../../styles/PatientEducation/patient-education.js';
+import color from '../../styles/color.js';
+
+const api = 'http://ec2-52-15-106-40.us-east-2.compute.amazonaws.com:8000';
```
Comment (jpatricia) [inline ChemoTracker/components/PatientEducation/Search.js:11]: once my branch is merged, should move this into a separate file under `/services` folder.. you can see my latest changes in my branch on the folder and the file

File: ChemoTracker/components/PatientEducation/Search.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+const header = {
+  'Accept': 'application/json',
+  'Content-Type': 'application/json'
+}
+
+export const getSearchResults = (searchQuery) => {
+  const queryparams = `query=${searchQuery}`;
+  return fetch(`${api}/patient-education/posts?${queryparams}`, {
+    method: 'get',
+    headers: {
```
Comment (jpatricia) [inline ChemoTracker/components/PatientEducation/Search.js:17]: lol what's the use of line 12 if you are not using it here
i.e `headers: header`

File: ChemoTracker/components/PatientEducation/Search.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+  handleClick = (item) => {
+    // Linking.openURL(item.url).catch(err => console.error('An error has occurred', err));
+    WebBrowser.openBrowserAsync(item.url);
+  }
+
+  renderSeparator = () => {
+    return (
+      <View
+        style={{
```
Comment (jpatricia) [inline ChemoTracker/components/PatientEducation/Search.js]: should separate this into a new style together with other styles

File: ChemoTracker/components/PatientEducation/Search.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        page: 1,
+        refreshing: true
+      },() => {
+        this.makeRemoteRequest();
+      }
+    );
+  };
+
+  handleLoadMore = () => {
+    // this.setState(
```
Comment (jpatricia) [inline ChemoTracker/components/PatientEducation/Search.js]: delete if not going to be used

File: ChemoTracker/components/PatientEducation/Search.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+class Search extends Component {
+	static navigationOptions = {
+		tabBarLabel: "Search",
+		tabBarIcon: () => (<FontAwesomeIcon size={24} name="search" color={color.navBarIcon} />)
+  }
+
+  searchQuery = this.props.navigation.state.params.searchQuery;
+  
+  state = {
```
Comment (jpatricia) [inline ChemoTracker/components/PatientEducation/Search.js]: should be inside constructor
i.e 
```
constructor(props) {
  super(props);
  this.state = {.......}
  // can access props right away by doing props.something
  // any other functions that you want to declare
  // e.g this.fn = this.fn.bind(this);
}
```
refer to my branch if confused, or google `constructor()` react

File: ChemoTracker/components/PatientEducation/Search.js [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+  });
+}
+
+class Search extends Component {
+	static navigationOptions = {
+		tabBarLabel: "Search",
+		tabBarIcon: () => (<FontAwesomeIcon size={24} name="search" color={color.navBarIcon} />)
+  }
+
+  searchQuery = this.props.navigation.state.params.searchQuery;
```
Comment (jpatricia) [inline ChemoTracker/components/PatientEducation/Search.js]: why not put it as state too?

File: ChemoTracker/styles/PatientEducation/patient-education.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+    marginRight: 20,
+    marginTop: 20,
+    padding: 10,
+    width: 130
+  },
+  subtitleText: {
+    color: "#A59494", 
+    fontSize: 14, 
+    marginTop: 10,
+    // marginRight: "5%",
```
Comment (jpatricia) [inline ChemoTracker/styles/PatientEducation/patient-education.js]: delete if not going to be used

**Review by jpatricia — COMMENTED** [review]
comments added

File: ChemoTracker/components/PatientEducation/PatientEducation.js (hunk 1 above)
Comment (lsivasothy) [reply ChemoTracker/components/PatientEducation/PatientEducation.js]: going to leave this for Forum

File: ChemoTracker/components/PatientEducation/Search.js (hunk 5 above)
Comment (lsivasothy) [reply ChemoTracker/components/PatientEducation/Search.js]: leaving this too for future to be able to load more results

File: ChemoTracker/components/PatientEducation/Search.js (hunk 3 above)
Comment (lsivasothy) [reply ChemoTracker/components/PatientEducation/Search.js:17]: removing it 😄

**Review by jpatricia — APPROVED** [review]

**Comment by jpatricia** [comment]
nice work :)

File: ChemoTracker/components/PatientEducation/PatientEducation.js (hunk 1 above)
Comment (shayywun) [reply ChemoTracker/components/PatientEducation/PatientEducation.js]: if its not hard to remember, i agree to remove it

File: ChemoTracker/components/PatientEducation/Search.js (hunk 2 above)
Comment (shayywun) [reply ChemoTracker/components/PatientEducation/Search.js:11]: maybe make a file (e.g. constants, or strings, etc) with the api in there and import that so its hidden from implementation

**Review by shayywun — APPROVED** [review]

File: ChemoTracker/components/PatientEducation/Search.js (hunk 2 above)
Comment (lsivasothy) [reply ChemoTracker/components/PatientEducation/Search.js:11]: I'll make a file once Janice's branch is merged :)

**Pull request merged**